### PR TITLE
Remove destroyed cache proxy from owner `CacheManager` on destroy

### LINKED ISSUE #8186: `ICache::destroy` should remove cache itself from the owner `CacheManager`
... otherwise it causes memory leaks because of zombie cache proxies in the `AbstractHazelcastCacheManager::caches`

### PR DESCRIPTION
**Pull request by serkan-ozal**
Fixes #8186

### CODE REVIEW

File: hazelcast-client/src/main/java/com/hazelcast/client/spi/ClientProxy.java [hunk 1]
```diff
@@ -98,16 +98,13 @@ public final void destroy() {
             context.removeProxy(this);
             try {
                 new ClientInvocation(getClient(), clientMessage).invoke().get();
+                postDestroy();
```
Comment (ahmetmircik) [inline hazelcast-client/src/main/java/com/hazelcast/client/spi/ClientProxy.java]: can it be more appropriate to put `postDestroy()` in a `finally` block?

Comment (serkan-ozal) [inline hazelcast-client/src/main/java/com/hazelcast/client/spi/ClientProxy.java]: I have just wanted it called only when invocation is successful. We may introduce another hook method for error case like `onDestroyError(Exception e)`

**Comment by emrahkocaman** [comment]
👍

**Comment by ahmetmircik** [comment]
👍

**Comment by serkan-ozal** [comment]
verify

**Pull request merged**